core/domain/stats_domain.py

```diff
@@ -1351,3 +1352,115 @@ def validate(self):
             raise utils.ValidationError(
                 'calculation_output is too big to be stored (size: %d): %s' % (
                     sys.getsizeof(output_data), str(output_data)))
+
+
+class LearnerAnswerDetails(object):
+    """Domain object that represents the answer details submitted by the
+    learner.
+    """
+
+    def __init__(
+            self, state_reference, entity_type, interaction_id,
+            learner_answer_info_list, schema_version=(
+                feconf.CURRENT_LEARNER_ANSWERS_DETAILS_SCHEMA_VERSION)):
+        """Constructs a LearnerAnswerDetail domain object.
+
+            Args:
+                state_reference: str. The reference to the state.
```

Comment:
I have mentioned this before -- this sort of docstring description is not useful. Please avoid just restating the name of the variable/argument/property in your description, and be more explicit/descriptive about what the possible values are and what they represent. Developers won't know what a "state reference" is, in general.

Please bear this in mind in the future.

Reply:
Done